Src/VimMac/StatusBar.cs

```diff
@@ -0,0 +1,168 @@
+                {
+                    oneTimeArgument = "insert";
+                }
+                else if (vimBuffer.InOneTimeCommand.Is(ModeKind.Replace))
+                {
+                    oneTimeArgument = "replace";
+                }
+            }
+            // Check if we can enable the command line to accept user input
+            string status;
+            switch (currentMode.ModeKind)
+            {
+                case ModeKind.Normal:
+                    status = string.IsNullOrEmpty(oneTimeArgument)
+                        ? string.Empty
+                        : string.Format("-- ({0}) --", oneTimeArgument);
+                    break;
+                case ModeKind.Command:
+                    {
+                        var command = vimBuffer.CommandMode.EditableCommand;
+                        return new EditableCommand(":" + command.Text, command.CaretPosition + 1);
+                    }
+                case ModeKind.Insert:
+                    status = "-- INSERT --";
+                    break;
+                case ModeKind.Replace:
+                    status = "-- REPLACE --";
+                    break;
+                case ModeKind.VisualBlock:
+                    status = string.IsNullOrEmpty(oneTimeArgument)
+                        ? "-- VISUAL BLOCK --"
+                        : string.Format("--({0}) VISUAL BLOCK --", oneTimeArgument);
+                    break;
+                case ModeKind.VisualCharacter:
+                    status = string.IsNullOrEmpty(oneTimeArgument)
+                        ? "-- VISUAL --"
+                        : string.Format("-- ({0}) VISUAL --", oneTimeArgument);
+                    break;
+                case ModeKind.VisualLine:
+                    status = string.IsNullOrEmpty(oneTimeArgument)
+                        ? "--VISUAL LINE--"
+                        : string.Format("--({0}) VISUAL LINE --", oneTimeArgument);
+                    break;
+                case ModeKind.SelectBlock:
+                    status = string.IsNullOrEmpty(oneTimeArgument)
+                        ? "-- SELECT BLOCK --"
+                        : string.Format("-- ({0}) SELECT BLOCK --", oneTimeArgument);
+                    break;
+                case ModeKind.SelectCharacter:
+                    status = string.IsNullOrEmpty(oneTimeArgument)
+                        ? "--SELECT--"
+                        : string.Format("--({0}) SELECT--", oneTimeArgument);
+                    break;
+                case ModeKind.SelectLine:
+                    status = string.IsNullOrEmpty(oneTimeArgument)
+                        ? "-- SELECT LINE --"
+                        : string.Format("-- ({0}) SELECT LINE --", oneTimeArgument);
+                    break;
+                case ModeKind.ExternalEdit:
+                    status = "External edit detected (hit &lt;Esc&gt; to return to previous mode, &lt;C-c&gt; to cancel external edit)";
```

Comment:
Should these be translated?

Reply:
They will be at some point once I/We have decided how to best handle shared translations between Wpf/Mac

Reply:
This whole class might disappear at some point